src/components/ThemeSwitch.tsx

```diff
@@ -0,0 +1,41 @@
+import { FC, useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { settingsActionCreator as setting } from '../actions/settings'
+import theme from '../selectors/theme'
+
+/** A ThemeSwitch Component with a title and description. */
+const ThemeSwitch: FC = () => {
+  const dispatch = useDispatch()
+  const currentTheme = useSelector(state => theme(state))
+  const [active, setActive] = useState(currentTheme === 'Light')
```

Comment:
It would be better to use Redux as the source of truth rather than creating duplicate state within the component. 

This will eliminate the need for the `useEffect`.